Mypy error with REST interceptors

In PR https://github.com/googleapis/python-datastore/pull/419, there is a mypy failure for the autogenerated code. See build log [here](https://github.com/googleapis/python-datastore/actions/runs/4198286941/jobs/7281708655).

```
google/cloud/datastore_v1/services/datastore/transports/rest.py:322: error: No return value expected  [return-value]
google/cloud/datastore_v1/services/datastore/transports/rest.py:345: error: No return value expected  [return-value]
google/cloud/datastore_v1/services/datastore/transports/rest.py:1401: error: "pre_cancel_operation" of "DatastoreRestInterceptor" does not return a value  [func-returns-value]
google/cloud/datastore_v1/services/datastore/transports/rest.py:1464: error: "pre_delete_operation" of "DatastoreRestInterceptor" does not return a value  [func-returns-value]
google/cloud/datastore_admin_v1/services/datastore_admin/transports/rest.py:276: error: No return value expected  [return-value]
google/cloud/datastore_admin_v1/services/datastore_admin/transports/rest.py:299: error: No return value expected  [return-value]
google/cloud/datastore_admin_v1/services/datastore_admin/transports/rest.py:1163: error: "pre_cancel_operation" of "DatastoreAdminRestInterceptor" does not return a value  [func-returns-value]
google/cloud/datastore_admin_v1/services/datastore_admin/transports/rest.py:1[22](https://github.com/googleapis/python-datastore/actions/runs/4198286941/jobs/7281708655#step:5:23)6: error: "pre_delete_operation" of "DatastoreAdminRestInterceptor" does not return a value  [func-returns-value]
```

The issue is that the typing below is not correct.

In the code below, the parameter `response` has type `{{ signature.request_type }}` but it should be `{{ signature.response_type }}`

https://github.com/googleapis/gapic-generator-python/blob/120f19eef34ca3d722f079683222be99931363da/gapic/templates/%25namespace/%25name_%25version/%25sub/services/%25service/transports/rest.py.j2#L132-L139

Also, the return type for the code below doesn't match the actual objects returned

https://github.com/googleapis/gapic-generator-python/blob/120f19eef34ca3d722f079683222be99931363da/gapic/templates/%25namespace/%25name_%25version/%25sub/services/%25service/transports/rest.py.j2#L124-L130

--- a/gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2
+++ b/gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2
@@ -121,7 +121,9 @@ class {{ service.name }}RestInterceptor:
     {% endfor %}
 
     {% for name, signature in api.mixin_api_signatures.items() %}
-    def pre_{{ name|snake_case }}(self, request: {{signature.request_type}}, metadata: Sequence[Tuple[str, str]]) -> {{signature.response_type}}:
+    def pre_{{ name|snake_case }}(
+        self, request: {{signature.request_type}}, metadata: Sequence[Tuple[str, str]]
+    ) -> Tuple[{{signature.request_type}}, Sequence[Tuple[str, str]]]:
         """Pre-rpc interceptor for {{ name|snake_case }}
 
         Override in a subclass to manipulate the request or metadata
@@ -129,7 +131,9 @@ class {{ service.name }}RestInterceptor:
         """
         return request, metadata
 
-    def post_{{ name|snake_case }}(self, response: {{signature.request_type}}) -> {{signature.response_type}}:
+    def post_{{ name|snake_case }}(
+        self, response: {{signature.response_type}}
+    ) -> {{signature.response_type}}:
         """Post-rpc interceptor for {{ name|snake_case }}
 
         Override in a subclass to manipulate the response

